[0.5, javalib] GZIPInputStream doesn't seem to be working correctly

When running same code on JVM and Native, I get this error on native:

```
Exception in thread "main"java.io.IOException
        at java.io.IOException.<init>(Unknown Source)
        at java.util.zip.InflaterInputStream.read(Unknown Source)
        at java.util.zip.GZIPInputStream.read(Unknown Source)
        at ExtractTar$.transferN$1(Unknown Source)
        at ExtractTar$.readObject$1(Unknown Source)
        at ExtractTar$.extract(Unknown Source)
        at ExtractTar$package$.hello(Unknown Source)
        at hello.main(Unknown Source)
        at <none>.main(Unknown Source)
Caused by: java.util.zip.DataFormatException: -3
        at java.util.zip.Inflater.inflateImpl(Unknown Source)
        at java.util.zip.Inflater.inflate(Unknown Source)
        ... 8 more
```

What follows is by no means a minimal reproduction, but a reproduction nonetheles.

The code attempts to decompress and read a GZIP-compressed tar archive and print out its entries.

Run on JVM: `scala-cli ExtractTar.scala`
Run on Native: `scala-cli ExtractTar.scala --native`

`ExtractTar.scala`
```scala
//> using scala 3.5.0
//> using nativeVersion 0.5.5

import java.io.*
import java.nio.file.*
import scala.util.boundary
import language.experimental.saferExceptions

import java.util.zip

import java.io.*
import java.util.zip.GZIPInputStream
import java.util.Base64

@main def hello(in: String*) =

  val is =
    in.headOption match
      case None =>
        val bytes =
          val dec = Base64.getDecoder()
          dec.decode(SAMPLE_TAR_GZ.trim())

        new GZIPInputStream(
          new ByteArrayInputStream(bytes)
        )
      case Some(value) =>
        new GZIPInputStream(new FileInputStream(File(value)))

  ExtractTar.extract(is, Paths.get("./out"))

object ExtractTar:
  def extract(tarStream: InputStream, out: Path) =
    boundary:
      def readObject() =
        var read = 0
        def ascii(n: Int) =
          read += n
          val bytes = tarStream.readNBytes(n)
          if bytes.length != n then boundary.break()
          else new String(bytes.takeWhile(_ != 0)).trim()
        val name = ascii(100)
        if name == "" then boundary.break()
        val mode = ascii(8)
        val ownerId = ascii(8)
        val groupId = ascii(8)
        val sizeStr = ascii(12)
        val mtime = ascii(12)
        val checksum = ascii(8)
        val linkOrType = ascii(1)
        val linkedFileName = ascii(100)
        val ustarIndicator = ascii(6)
        val headerRemaining = 512 - read
        val isDir = linkOrType == "5"

        val size = Integer.parseInt(sizeStr, 8)

        val destination = out.resolve(name)

        val skipped = tarStream.skip(headerRemaining)

        val padding = if size % 512 == 0 then 0 else (512 - (size % 512))

        if isDir then 
          println(s"Creating directory $destination")
        else 
          println(s"Would be writing file $destination, size $size")
          tarStream.skip(size + padding)

      end readObject

      while true do readObject()
  end extract
end ExtractTar

val SAMPLE_TAR_GZ =
  """
H4sIAG0qFWcAA+1Y2c7jWlYukBDi5w4JrtOlRvp/XB3PQw46CDtxnMlD4tmt1pHnIZ7i2UG8B/c8ChJPwS1vwAPg/KkhVdXVzcVpWohaUhR7r/Wtvdbaa9jJ/Cd2aCrbbRS7mteundpvfnaCIIjAsNntmyTw128Iub9D92d0BmMkhCMIDuPYDIIJHIXezKCf35Svqa0bu5pM6fy08xvHTr0szvyv5SaxIPgdeu6uzD5+/1+hP/ubP3/zp2/e8LY7E+WZMXtPt7U3fzF9kOlzmT6393/9n6mkFeX0/vGG+Jfp85dfiPzJp/W/dotsbpdl6s/Lquj83M5d/ybwyzf//lf/+V//8W8/g5Pf6Vsk2cPGtz2/Av9wfeD31j8MfV7/CITgxJvZ8POZ8G36f17/KDTLmsnhH2ESWWAYhBLYHCbQ6cAQFH/Cydlhy9Cn5WarsfPBbppq/tvK9Uf6SNM7esWxrQGhPP+ELWbyBDqYvwv0UONPf+w4/H+lP/z0/331D6MwBn05/1EY/T7//zcozsqiamaJ3dnzuJj/3dPjQj6tBPFUtB+XX3Nk3jZxOneKNvfsavzASu08bO3Qn/tD6VdTDPPGTue1HfgVO7h+2cRFXj99pv9VzzUuP1/9yowPYnPO2krbvGwbual8O/taiLFrn8Cenv4xs+N85vnBLPLTtHiO8x9mEybOw5fZj0+zWWens7ie/TjL/X72hdbn29p68vpx7fY+aXl5eXma4A9F498fn+P63Uyym6ieh37z/HYOFm3z9iZdOInvNg+QHyYFN8s+IKf0u2/yw+xhx3ezScEPrypvJj/d7mQ31ITJ6ymkwrTjF4D89t5MwIkhts1Hzt3lG93cdtpggk+u01Vlj/MiWMXZr5mx8X/zDE9V+fJRtJpNYG8S/BX8sPZh/8q/saD3nD6a4jP7p6cPt8f3yLie356e73u+m0HvZs/57FePSl7mWZzfd375+zvuHz6q/ef3317xUXNRz/sqbvxHnTfUy0eJRwuBH1+ZT09fsV5X/Nz7FNC70Ies/uHB56C4pYr4msFzPyub8de3fHiM8W8+WDod0W1H8fXUnz+F/rOIQh8Xb/J27cbx8/30HgHvwzj5kD+svZ7hdFw3kz6mzmuYhdsp1s/5y4N0HNyF56mfh000+8WkbNZEfv7R0bkzYc/PjyA/rf3XyriXzPNdQ2Offf12zs8/3dRALy9PjzblduZPJt2dgSHoE3ey4c78cfb27e/Z/KYpK7xPmqjPeUWf+9XW+xY7rIq2/Da7jq/+5NMnM5Evdr7dhL7JdSPfPddt9i3taZyfxUoZywcVX0v43i13hG9G6yb3OpO2uRe7dlN8Mpf4XCp6vbef/Fuvm45pEsNhZKqu14x/FIzrVXzT8mjgdBb426evojNJTVnoh341L+2qnnpg8/w+au9m1MvnAM+vmzi3b2Ux4aZ+NaVhXUxj9Pl23l8I1+e4LF+bxqesva09f+HGF7DS9ry7c1MavVr4t69uTg5A91yC7vn6fHf++ZPMy4OqCXyPwg3yeX384nYe9fyVPdVsUY3PD469fImYze7y7uRB43/AxFPhPaI+Am62fVG993byYco8dpFHFfOmeB06L4+V+an7fwziu9eovLtpfZScXuduWtSTgqdHBZ/H/n10H0RAcDYNsEMR3lIgzoPiuX77fnZNh/frX95O9jdvP+x0a6Cf2t0HLfdp0FStP3Xuz9rh0x3yfvQ93Z4/Dcanp1t0ZJqXDuxPCn36ibNeu+Hbt2+fNli9pRn+stZdmgZgA7HJSmfYxXa5Zn1WXFqgd9qOBtZkLjmq6gZzB8Rq2l3unGwNIxaSLPtYnIqsBDIiDiyHddCPEdjTep9bVIg3CIWuSK/canvZ8ikbvhAXw1GrXsGh+KroiaaczLEfPBYAAWBgWVrZ8GyCV7lUAyp8TK/aANjm6Xw5KwdWHTxnL7JLZQPVvXfMB0C7EtmBzWHucJHpkdoQqoMqAgj3rkeoPDJ2oZSzYTWCEh4UFrjAWw+VSITkgJrjS/W0MrLzOQ/QJRgdKSFflfAVXCjACme0td/19SHAljkMAGqQ75YgaJzaKPGjE05vu5infWYTxaEmJSPZTva3Pgh2HHodVudEwdc8CZm0ALF8QK7QoxQgIHWmnArcm4f4smrPwKZvg10FXMHpK6InC1jwvNEvo0L3C0PnK43nT1djZxV6sx3FaFC7JnRWtVnofplsEMdHiZ5f4EZ7JEGdyNjghBkcZ8f0KJwuXcdX1YVyL1JuBqAGVlEJgugiBzzLjFEKB7JW6CgQAY4E5EX5cudInjsI1AnQyDRjdkzSOkyZL/Kcq7kRlvRescJ1GWFBDsn+poktJ+T5NcNtzSOx0iS/Px08Gl1CPLvXWm4paxoY5fmOdzrfAnDeVrjGCpZXPooPO1XJxBAYqKoJjqzLjil19gKvl1iGcWKapw5n092HrUxEyCju1US8UJ3N0M4YrseaZ9opTEvWZSzdRlxhzxs8T7gXCpCsi9QwmK1KoKLrfbe8NFqVsp6+PrMhvapDBySKtOp1FBcFU4KxwzXHXXqP2ey6By4IOa7KvLxYonpIjY5XrjzUNbK4GOrQWvuZM3Zb0S3ycYj22gLxtwvXCVwxPK66K07DnDQkDmMuHXPLbmqKJbuE2y4TY6/IbJhlunRuLgmVkgrnZUeDN8FVSh8OR6LqFjAX0Kh22iG+CFuCaof18ax1477zWZAcCFplTUAOe75adamC+5lB7MU0Q1apBuojfm1Zf+upNCBXNk7beNUxlh/Y5LhAAjTmwEBKkH4Mzi7PitSpXV6wQSY6u57KMQKSla7FZbA8Ei7EhF2sQEWnqFsf6MzKWdZpnOEISYcMkA7EguQFstpswCu+PFNFftgQQa5eRqAMcO2ElalFtO0OcyptBFKPu44Z65pHQ+JaPdzhh6m4sjhI97KjFRs7JbXCsNx+FVoQMTLuccWePXbVb+IMCUkbJ5yjTDG7gaSxPRkayGTjVfMgurmSOVGYx8iujH2jAgwZSgUUq+Y5ZvxSWG5ikGZaBwTzGj7sGAohr4KMkWot0cdwXQ9S7EPYdb+T4kFCGY7hDwguO6jZBrC2ULMC64UwzQKq9HhK8BrBFFXLYJlFRR/EOnfPk1rjkvu+w+7itrCDKCY71x7RHGQWNN3qi6VJ7S4Ve2acrjKkjRFnpOyqrWr3QaQ7KzQ1gZEmtXB93DrDOj4PfUbHQJnADRYlXlnRA3++nLrC0hpaRKOgV3b8MYnlC6DhHbwWvQxTcxkQUqKYZoeKmk4/xKiidPRxMwJJjemgZLpbOmbrhrU8otKIZUWSBbwnwkJi0yVh7AiQxIpklJhVezBdgKOcNbHWkqY66ktDOgocy+7zbHCRTWJoNcWV+qJrO2/MLWWxCEy9NR0aqmhVQzojD9IRuFI55UH4HljsYa4RWv7IH6A+PIqqjW2MdkMKZGkekiUzhMWCbmWKFqrD0kZXoHjII8xAF1be9ugCosj0FBQjaqDRZm8xWebSRnTtyELcD74yLNbG3lkETCamaubksHcEUMxBF6Cv0Ra/DTaUttDkFQ6v5dIzCC2rhPJ66ERJL0xMavTIcxBjPQhLD5juuy2AM9mwpy33cvSIkiOKlYQL64wJQiFdc6myEg6K7cUwR0CCLTHXXKS67EIuk9jsTaK46vvNSvOiMyFvMcuPMVNFnFhjt30VluKhTg9osmoLjrQFsraO+1ZX5H59TYQk31I4xkxJctphRxneTdVXQ6xgXw4Y4jm7BNWVaRJcBQdMZGtVZskpsjSVOgmtj3idIK/J7swjzVq8eFuMSJxAYt04A8ps11wDkTkdwhPeD4JsU5VAi1USOT3UH1JXuqoaeCAaIY58tNWEa4FLqW/sT76y0W1qDw/lNlkOfFThQcJUFCUcN/YpoyEvh9CBc1GDHPfBoTqAdnxG8Awrd6hdSqSu+jgX9uJiMzVToQLO8gnS0+hSj0ydM4R50mVoby2SqSSlHYxBFX7A/HFpAAeyLW2rGBPetY4DZa03GNMLNue6iQpsAl0QTlWIxwanoOJSv4gGXBvWhefNM5LTjawunf2BhFODRTI4rJGyVtcZGtWKDhNdaiPUERttM+s3HKWQ6h7FwUPrXU/6RRhgbJ/DS2zf6GoAECk6bpcNM3GHolcSXNts4XV5tTXLjXZoYK4VcA2Dy+xkEVV6RS/Lsm9xYWCniSsIAnSRjC19PsMrhYrlxdrmeLQn+aHKojhwQu00Nknfpw1PY0lwHlkjB05LVW3rTT+MuIbvhsRSYcXYuazeGye6WV+JgoOgS+YWiw1RbdJk+hm6hvUtsmNPZwSK8YCdLjBslOKt0Kva8jI1pyOGoLEAGL18UG0njZBtNO5Vvt2rZbiFw+lmdryuD7XCTj1hrznXbGNwKzsclcrLruJ2UXmtomzxrZSTKefq8mUXAQusaozlCYNd+6AxJ1yAbMoe6TomYK5qhShwLD0QawLl3MuyMEuMswSvIreber2APR9yOiWCucSMV3k/xiYqCYoQihkDOKiMMu1W0ExI9NXE6V1c0ioVpW0/2bOwv68UW8Qa4rDuNMDEN3zMW2e7xq4ZzuKQodeYuvFF4gLCZqVy1siiF6b0N0IsrHvjuHD3ixYPJTGJMDzrz5IrC4aEAmAhDzyLIUjd1o7nhjKglytVaUc/Rf2zp0W9tsyaMl3CqVycObAc3EUUNcJSIi4QQYnZKBVGCulRD6YXndlhW72mmubgkTh53nULcl9ur3ZtuYCGKMy6QDL3eu3LmqnW8ondwRlCLvQwneIrLNeXZoug/MGEnMoSzLLqRj4tT4e9vtt4sHul4pUArgyzYnJ4ZbfTJdZz/elGvKR62OV28f4wuIIyJiKM2DaBDCfC23VrFl4K+cknnVUUnDYuCFk7uk5bRXWFqWwGjGNqlBtRItLTgWxPPV9U6dQJdzsH9opIMFBG92wNYOJ6dwiVUymtKc6eWi5KYWS3GPx6y+I4jNeIvENyeU8yoyQgdcbrRBXCBBnh445fSKEDbBoI7UFzUCnt6JmJwKdGEisW2nSWnVK+UqeijMtMuDKDoPFWJ6PiM40hHe906zt2dBb9rFlfauFoeiSDtKhS7/ewZgipwDRXDYSZ9bmvdrjskyyCbg5Oe91G/hXmRnE5Tj9kyjPQbuup866R8Qh5l/4a71zHFxNIALdVsrM4+LIkvJzb7XsMzbhEm35QnZGQQdGiRDnyeoJ77Uhq9Vrie2O5xc9lZkHC9VRXB0MARYWlTIKWBFHM8vNyZ4O5nO5DJuUOfOLLTFCvr/UU/pL1ZR7JBR0FrUI6tlvAw3aNTErR9SpJsLI3AyEBSQCETLj19ryD+hE3XbNOiY5DxdB5l/0JpAffOosqvadp+un2w+2P/f/2d/pO3+k7faffTv8NBna2MgAmAAA=
"""

```

--- a/javalib/src/main/scala/java/util/zip/InflaterInputStream.scala
+++ b/javalib/src/main/scala/java/util/zip/InflaterInputStream.scala
@@ -108,16 +108,19 @@ class InflaterInputStream private (
     }
   }
 
+  /* Fix Issue 4071. Make only required minimal changes so as to
+   * not disturb the dead hand of the past.
+   */
+
   override def skip(nbytes: Long): Long = {
-    if (nbytes >= 0) {
-      if (buf == null) {
-        buf =
-          new Array[Byte](Math.min(nbytes, InflaterInputStream.BUF_SIZE).toInt)
-      }
+    if (nbytes >= 0L) {
+      val skipBuffer =
+        new Array[Byte](Math.min(nbytes, InflaterInputStream.BUF_SIZE).toInt)
+
       var count, rem: Long = 0L
       while (count < nbytes) {
         val x = read(
-          buf,
+          skipBuffer,
           0,
           if ({ rem = nbytes - count; rem > buf.length })
             buf.length

